RefreshTests.GameServer/Tests/Levels/ScoreLeaderboardTests.cs

```diff
@@ -394,4 +395,60 @@ public void DoesntPlayLevelWithCountOnMainline()
         HttpResponseMessage message = client.PostAsync($"/lbp/play/user/{level.LevelId}?count=3", new ReadOnlyMemoryContent(Array.Empty<byte>())).Result;
         Assert.That(message.StatusCode, Is.EqualTo(BadRequest));
     }
+
+    [Test]
+    public void OvertakeNotificationsWorkProperly()
+    {
+        using TestContext context = this.GetServer();
+        const int userAmount = 10;
+
+        List<GameUser> users = Array.Empty<GameUser>().ToList();
+        for (int i = 0; i < userAmount; i++)
+        {
+            users.Add(context.CreateUser());
+        }
+        
+        GameLevel level = context.CreateLevel(users.First());
+
+        for (int i = 0; i < users.Count; i++)
+        {
+            GameUser user = users[i];
+            context.SubmitScore(i, 1, level, user, TokenGame.LittleBigPlanet2, TokenPlatform.PS3);
+
+            // Check that notification was sent to the last #1 users
+            if (i <= 0) continue;
+
+            GameUser notificationRecipient = users[i - 1];
+
+            bool notificationSent = false;
+            foreach (GameNotification notification in notificationRecipient.Notifications)
+            {
+                if (notification.Title != "Score overtaken") continue;
+                
+                notificationSent = true;
+                context.Database.DeleteNotification(notification);
+            }
+            
+            if (i - 1 == 0) // Notification should not be sent if the last #1 entry had a score of 0
+                Assert.That(notificationSent, Is.EqualTo(false));
+            else
+                Assert.That(notificationSent, Is.EqualTo(true));
```

Comment:
Having assertions behind flags like this is a little sketch, I won't lie. Can you add checks to make sure that these are both run?

Reply:
Would replacing that with the following suffice?
`Assert.That(notificationSent, Is.EqualTo(i - 1 != 0));`

Reply:
Right, gotcha